config/deploy.rb

```diff
@@ -1,15 +1,23 @@
 # config valid only for current version of Capistrano
-lock "~> 3.16.0"
+lock "~> 3.17.1"
 
-def deploysecret(key)
+def deploysecret(key, default: "")
   @deploy_secrets_yml ||= YAML.load_file("config/deploy-secrets.yml")[fetch(:stage).to_s]
-  @deploy_secrets_yml.fetch(key.to_s, "undefined")
+  @deploy_secrets_yml.fetch(key.to_s, default)
+end
+
+def main_deploy_server
+  if deploysecret(:server1) && !deploysecret(:server1).empty?
+    deploysecret(:server1)
+  else
+    deploysecret(:server)
+  end
 end
 
 set :rails_env, fetch(:stage)
 set :rvm1_map_bins, -> { fetch(:rvm_map_bins).to_a.concat(%w[rake gem bundle ruby]).uniq }
 
-set :application, "participacyl"
+set :application, deploysecret(:app_name, default: "consul")
```

Comment:
I'm wondering whether it's easier to leave it like `set :application, deploysecret(:app_name, default: "participacyl")` so we don't have to edit our `deploy-secrets.yml` file when we deploy to this repo :thinking:.